controller/tests/controller_server/csi_controller_server_test.py

```diff
@@ -1195,6 +1199,156 @@ def test_controller_get_capabilities(self):
+        self.mediator.expand_volume.assert_called_once_with(volume_id=self.volume_id,
+                                                            required_bytes=self.capacity_bytes)
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_expand_volume_succeeds_block_access_mode(self, storage_agent):
+        self._prepare_expand_volume_mocks(storage_agent)
+
+        self.request.volume_capability = csi_pb2.VolumeCapability(
+            access_mode=csi_pb2.VolumeCapability.AccessMode(mode=self.access_types.SINGLE_NODE_WRITER),
+            block=csi_pb2.VolumeCapability.BlockVolume())
+
+        expand_respond = self.servicer.ControllerExpandVolume(self.request, self.context)
+        self.assertEqual(self.context.code, grpc.StatusCode.OK)
+        self.assertTrue(expand_respond.node_expansion_required)
+        self.mediator.expand_volume.assert_called_once_with(volume_id=self.volume_id,
+                                                            required_bytes=self.capacity_bytes)
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_expand_volume_with_wrong_secrets(self, a_enter):
+        self._test_create_object_with_wrong_secrets(a_enter)
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_expand_volume_with_wrong_volume_capabilities(self, storage_agent):
+        storage_agent.return_value = self.storage_agent
+
+        self.request.volume_capability = csi_pb2.VolumeCapability(
+            access_mode=csi_pb2.VolumeCapability.AccessMode(mode=self.access_types.SINGLE_NODE_WRITER),
+            mount=csi_pb2.VolumeCapability.MountVolume(fs_type="ext42"))
+
+        self.servicer.ControllerExpandVolume(self.request, self.context)
+        self.assertEqual(self.context.code, grpc.StatusCode.INVALID_ARGUMENT, "wrong fs_type")
+        self.assertTrue("fs_type" in self.context.details)
+
+        self.request.volume_capability = csi_pb2.VolumeCapability(
+            access_mode=csi_pb2.VolumeCapability.AccessMode(mode=self.access_types.MULTI_NODE_SINGLE_WRITER),
+            mount=csi_pb2.VolumeCapability.MountVolume(fs_type="ext4"))
+
+        self.servicer.ControllerExpandVolume(self.request, self.context)
+        self.assertEqual(self.context.code, grpc.StatusCode.INVALID_ARGUMENT, "wrong access_mode")
+        self.assertTrue("access mode" in self.context.details)
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_expand_volume_with_array_connection_exception(self, storage_agent):
+        self._test_create_object_with_array_connection_exception(storage_agent)
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_expand_volume_with_get_volume_name_too_long_success(self, storage_agent):
+        self._prepare_expand_volume_mocks(storage_agent)
+        self.mediator.max_volume_name_length = 63
+
+        self.request.name = "a" * 128

```

Comment:
where is this being used?

Reply:
that test is not relevant and I removed it completely